client/my-sites/customer-home/components/full-screen-launchpad.tsx

```diff
@@ -38,13 +41,21 @@ export const FullScreenLaunchpad = ( { onClose }: { onClose: () => void } ): JSX
 	} );
 
 	const onSiteLaunched = async () => {
-		await updateLaunchpadSettings( siteId, {
-			checklist_statuses: { site_launched: true },
-		} );
+		setIsLaunching( true );
+		launchSiteApi(
+			async () => {
+				await updateLaunchpadSettings( siteId, {
+					checklist_statuses: { site_launched: true },
+				} );
 
-		await refetch?.();
-		layout?.refetch();
-		dispatch( requestSite( siteId ) );
+				await refetch?.();
+				await layout?.refetch();
+				await dispatch( requestSite( siteId ) );
+				onClose();
+				setIsLaunching( false );
```

Comment:
We should wrap this async function in a `try {} finally {}` so we know the flag gets reset even if there's a network error.

Reply:
Good catch! [Fixed it here](https://github.com/Automattic/wp-calypso/pull/99563/commits/331b43b8541b2fc9f819413aa71545a081e3edc7).